### PR TITLE
Upgrade jblas

### PR DESCRIPTION
**Pull request by luoyu-intel**
## Type of Change
Upgrade jblas: a more flexible dynamic-quant interface.

## Description

1. Synchronize jblas code.
2. Remove the high gcc version requirement. 
3. auto-fusion: depends on weight type and runtime ISA support.

## How has this PR been tested?

gptj-6B and llama-7B model.

### CODE REVIEW

File: intel_extension_for_transformers/backends/neural_engine/graph/jblas/jblas/jit_blas_weight_compression.h [hunk 1]
```diff
@@ -1105,63 +1217,78 @@ namespace gemm_default {
...
+        jblas::epilogue::gemm::AccumulatorWriteBackFp32>,  // output fp32->fp32
     DefaultParallel>;
 }  // namespace amx_bf16
 namespace amx_int8 {
 JBLAS_ISA constexpr DefaultISA = JblasAMX_INT8;
-using GemmSKernelDynamicS4KBlock = jblas::wrapper::gemm_kblock::GemmInterfaceKBlockPackWeight<
+using GemmSKernelDynamicS4ClipFp32KBlock = jblas::wrapper::gemm_kblock::GemmInterfaceKBlockPackWeight<
+    jblas::wrapper::gemm_kblock::GemmSLauncherKBlockPackWeight<
+        DefaultISA, jblas::gemm::kblock::GemmCore_Row_NN_16x48_AMX_INT8_KBLOCK,
+        jblas::prologue::gemm::ActivationF32S8KBlockQuantize,
```
Comment (zhewang1-intc) [inline intel_extension_for_transformers/backends/neural_engine/graph/jblas/jblas/jit_blas_weight_compression.h:1273]: seems that we don't update the `prologueA` template with `ActivationFp32SymS8Quantize`, will we replace these old templates like `ActivationF32S8KBlockQuantize` in weight_compression in the future? can `ActivationFp32SymS8Quantize` support dynamic_quant which amx_int8/vnni gemmcore needed?

Comment (luoyu-intel) [reply intel_extension_for_transformers/backends/neural_engine/graph/jblas/jblas/jit_blas_weight_compression.h:1283]: ActivationF32S8KBlockQuantize is the KBLOCK quantize template, it binds with KBLOCK GemmCore. ActivationFp32SymS8Quantize is the per-channel quantize template, it binds with non-KBLOCK GemmCore.

**Review by DDEle — APPROVED** [review]
LGTM

File: intel_extension_for_transformers/backends/neural_engine/graph/jblas/CMakeLists.txt [hunk 2]
```diff
@@ -24,5 +29,5 @@ set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
 include(FindOpenMP)
 target_link_libraries(${PROJECT_NAME} INTERFACE OpenMP::OpenMP_CXX OpenMP::OpenMP_C)
-target_compile_features(${PROJECT_NAME} INTERFACE cxx_std_11)
+target_compile_features(${PROJECT_NAME} INTERFACE cxx_std_17)
```
Comment (DDEle) [inline intel_extension_for_transformers/backends/neural_engine/graph/jblas/CMakeLists.txt:32]: Is it problematic if a head-only library uses newer standard then that the project depends on it?

cc @airMeng

Comment (luoyu-intel) [reply intel_extension_for_transformers/backends/neural_engine/graph/jblas/CMakeLists.txt:32]: cxx17 is not that new, only gcc>=7 should be alright. BTW, cxx26 is under development.

**Review by zhewang1-intc — APPROVED** [review]
LGTM

File: intel_extension_for_transformers/backends/neural_engine/graph/models/model_utils/model_utils.cpp [hunk 3]
```diff
@@ -767,43 +767,99 @@ quant_params_internal quant_params_to_internal(const quant_params& params) {
...
-      using GemmKernel = jblas::wrapper::gemm_default::weight_comp::avx512f::GemmKernelS4KBlock;
-      static GemmKernel kernel;
-      assert(cd->AVX512F());
-      packedw = kernel.getWeightPtr()->compressWeightTranspose(n, k, f32ptr, k, params.block_size, type);
-    } else if (params.compute_type == quant_comp::bf16) {
-      using GemmKernel = jblas::wrapper::gemm_default::weight_comp::amx_bf16::GemmKernelS4KBlock;
-      static GemmKernel kernel;
-      assert(cd->AVX512F());
-      packedw = kernel.getWeightPtr()->compressWeightTranspose(n, k, f32ptr, k, params.block_size, type);
+    if (params.scale_dtype == quant_sdtype::fp32) {
```
Comment (airMeng) [inline intel_extension_for_transformers/backends/neural_engine/graph/models/model_utils/model_utils.cpp:788]: ```else``` gone....

**Pull request merged**